BibBuddy/app/src/main/java/de/bibbuddy/BookFragment.java

```diff
@@ -52,8 +51,11 @@
   private BookRecyclerViewAdapter adapter;
   private List<BookItem> selectedBookItems;
 
+  DatabaseHelper dbHelper;
+  BookDao bookDao;
```

Comment:
DatabaseHelper and BookDao should not be used here. The BookModel already use them. Create the methods were you used the dbHelper and bookDao in the BookModel.

Reply:
In this case, we need a get-Methods for the bookDao and dbHelper in the BookModel-class. Just because the bookDao is declared as private final and databaseHelper is declared and initialized only in the constructor.

Reply:
Okay I fixed it. I created get-Methods to get the bookDao. But I need to create the dbHelper for the noteDao (an object of the class NoteDao.java). Have you any suggestions?

Reply:
ok, I fixed it.